library/alloc/src/alloc.rs

```diff
@@ -247,6 +257,8 @@ unsafe impl Allocator for Global {
     }
 
     #[inline]
+    #[requires(layout.size() == 0 || layout.align() != 0)]
+    #[requires(ptr.as_ptr() as usize % layout.align() == 0)]
```

Comment:
don't we need to require that the pointer is actually allocated?

Reply:
We actually need an even stronger precondition (and Kani correctly detects this via the `grow_impl` harness): we'd need to express that `ptr` either points to a ZST (that's doable) or is a valid heap allocation (I don't know how we could currently express this).

Reply:
We would need ghost state. :smile: